actors/builtin/miner/miner_test.go

```diff
@@ -2052,15 +2052,129 @@ func TestWithdrawBalance(t *testing.T) {
 
 func TestChangePeerID(t *testing.T) {
 	periodOffset := abi.ChainEpoch(100)
+	actor := newHarness(t, periodOffset)
+	builder := builderForHarness(actor).
+		WithBalance(bigBalance, big.Zero())
 
 	t.Run("successfully change peer id", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		newPID := tutil.MakePID("test-change-peer-id")
+		actor.changePeerID(rt, newPID)
+	})
+}
+
+func TestCompactPartitions(t *testing.T) {
+	periodOffset := abi.ChainEpoch(100)
+	actor := newHarness(t, periodOffset)
+	builder := builderForHarness(actor).
+		WithBalance(bigBalance, big.Zero())
+
+	t.Run("successfully compact partitions", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.SetEpoch(200)
+		info := actor.commitAndProveSectors(rt, 2, defaultSectorExpiration, [][]abi.DealID{{10}, {20}})
+
+		st := getState(rt)
+		dlIdx, partIdx1, err := st.FindSector(rt.AdtStore(), info[0].SectorNumber)
+		require.NoError(t, err)
+		_, partIdx2, err := st.FindSector(rt.AdtStore(), info[1].SectorNumber)
+		require.NoError(t, err)
+
+		// add partitions to state
+		partitions := bitfield.NewFromSet([]uint64{partIdx1, partIdx2})
+
+		actor.compactPartitions(rt, dlIdx, partitions)
+	})
+
+	t.Run("fails if deadline is equal to WPoStPeriodDeadlines", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.compactPartitions(rt, miner.WPoStPeriodDeadlines, bitfield.New())
+		})
+	})
+
+	t.Run("fails if deadline is not mutable", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		epoch := abi.ChainEpoch(200)
+		rt.SetEpoch(epoch)
+		rt.ExpectAbort(exitcode.ErrForbidden, func() {
+			actor.compactPartitions(rt, 1, bitfield.New())
+		})
+	})
+
+	t.Run("fails if partition count is above limit", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		// partition limit is 4 for the default construction
+		bf := bitfield.NewFromSet([]uint64{1, 2, 3, 4, 5})
+
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			actor.compactPartitions(rt, 1, bf)
+		})
+	})
+}
+
+func TestCheckSectorProven(t *testing.T) {
+	periodOffset := abi.ChainEpoch(100)
+
+	t.Run("successfully check sector is proven", func(t *testing.T) {
+		actor := newHarness(t, periodOffset)
+		rt := builderForHarness(actor).
+			WithBalance(bigBalance, big.Zero()).
+			Build(t)
+		actor.constructAndVerify(rt)
+
+		sectors := actor.commitAndProveSectors(rt, 1, defaultSectorExpiration, [][]abi.DealID{{10}})
+
+		actor.checkSectorProven(rt, sectors[0].SectorNumber)
+	})
+
+	t.Run("fails is sector is not found", func(t *testing.T) {
+		actor := newHarness(t, periodOffset)
+		rt := builderForHarness(actor).
+			WithBalance(bigBalance, big.Zero()).
+			Build(t)
+		actor.constructAndVerify(rt)
+
+		rt.ExpectAbort(exitcode.ErrNotFound, func() {
```

Comment:
Forwarding advice from @acruikshank, `ExpectAbortContainsMessage` is a great way to make sure you're getting the _right_ error.

Reply:
Will use this going ahead.